test/arrayops.jl

```diff
@@ -1705,6 +1705,30 @@ end
     @test istriu([1 2 0; 0 4 1])
 end
 
+#issue 49021
+@testset "reverse cartesian indices" begin
+    @test isequal(reverse(CartesianIndices((2, 3))), CartesianIndices((2:-1:1, 3:-1:1)))
+    @test isequal(reverse(CartesianIndices((2:5, 3:7))), CartesianIndices((5:-1:2, 7:-1:3)))
+    @test isequal(reverse(CartesianIndices((5:-1:2, 7:-1:3))), CartesianIndices((2:1:5, 3:1:7)))
```

Comment:
`isequal` for `AbstractArray` is element-wise. We'd better add a type assertion
```julia
@test isequal(reverse(CartesianIndices((2, 3)))::CartesianIndices, CartesianIndices((2:-1:1, 3:-1:1)))
```
or use `===` instead.
```julia
reverse(CartesianIndices((2, 3))) === CartesianIndices((2:-1:1, 3:-1:1))
```